_adr/87/index.adoc

```diff
@@ -1,48 +1,452 @@
+A list of the metrics retrieved by KAS Fleet Manager can be found in the https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/client/observatorium/api.go[observatorium] client within the project. When new metrics are added to this list, they become available in the KAS Fleet Manager _/metrics/query_ and _/metrics/query_range_ endpoints. In order to make new metrics available to the _/metrics/federate_ endpoint, their metadata must be added to this https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/constants/metrics.go[file] in KAS Fleet Manager.
+
+KAS Fleet Manager controls the labels exposed for each metric returned in all endpoints. 
+Metrics retrieved from Observatorium are also filtered so that end users will only receive metrics for their own Kafka instance.
+Apart from these, no additional aggregation, filtering or modification of the metric is done by KAS Fleet Manager.
 
 ## Proposal
-// What is the decision being proposed
+### Metric and Label Naming
+All metric and label names must adhere to the policy stated in link:../../_ap/16/index.adoc[AP-16: Naming Policy for Customer Facing Metrics].
+
+### Kafka Metric Categories
+The customer facing Kafka metrics can be divided into multiple categories.
+Categorizing these metrics into logical groups can help us formulate the metric name and document it.
+
+|===
+|Category | Description | Sub-component (to be specified in the metric name)
+
+|Cluster
+|General Kafka metrics not specific to any other sub-components.
+|cluster
+
+|Broker
+|Metrics associated to Kafka brokers
+|broker
+
+|Topic
+|Metrics associated to Kafka topics
+|topic (Even though the topic metrics already have a 'topic' label, including this improves the clarity of the metric name)
+|===
+
+Examples of sub-component usage in the metric name:
+
+* For *Cluster* metrics, the metric prefix would be `kas_cluster_`.
+* For *Broker* metrics, the metric prefix would be `kas_broker_`.
+* For *Topic* metrics, the metric prefix would be `kas_topic_`.
+
+### Converted Metrics
+At minimum, each metric should have an `instance_name` label that identifies the Kafka instance if possible.
+This would be useful if a user has multiple Kafka instances.
+If they want to integrate all of their Kafka metrics into one monitoring system, the `instance_name` can act as an identifier.
+
+#### Cluster Metrics
+|===
+|Old Metric Name | Revised Metric Name | Type | Old Labels | Revised Labels | Description
+
+|kafka_instance_max_message_size_limit
+|kas_cluster_message_size_limit_bytes
+|Gauge
+|Not available
+|instance_name
+|Maximum message size, in bytes, supported by the instance.
+
+|kafka_instance_partition_limit
+|kas_cluster_partition_limit
+|Gauge
+|Not available
+|instance_name
+|Maximum number of partitions per instance
+
+|kafka_namespace:haproxy_server_bytes_in_total:rate5m
```

Comment:
FYI - there is an defect affecting the implementation of this metric at the moment, tha means the values produce by it and `kafka_namespace:haproxy_server_bytes_out_total:rate5m` can be incorrect. https://issues.redhat.com/browse/MGDSTRM-9131

Reply:
That defect is fixed in production.

Reply:
Thanks for the update on this @k-wall